api.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import copy
+from pathlib import Path
+import sys
+from typing import Dict
+from unittest.mock import patch
+
+from fastapi import Request, FastAPI, status
+from fastapi.middleware.cors import CORSMiddleware
+import graphviz
+import networkx as nx
+import uvicorn
+
+from auth.auth import authenticate
+from wic import ast, cli, compiler, inference, labshare, utils, plugins, __version__  # , utils_graphs
+from wic.schemas import wic_schema
+from wic.wic_types import GraphData, GraphReps, Json, NodeData, StepId, YamlTree
+
+
+app = FastAPI()
+
+origins = ["*"]
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+def get_args(yaml_path: str = '') -> argparse.Namespace:
+    """This is used to get mock command line arguments.
+
+    Returns:
+        argparse.Namespace: The mocked command line arguments
+    """
+    testargs = ['wic', '--yaml', yaml_path, '--cwl_output_intermediate_files', 'True']  # ignore --yaml
+    # For now, we need to enable --cwl_output_intermediate_files. See comment in compiler.py
+    with patch.object(sys, 'argv', testargs):
+        args: argparse.Namespace = cli.parser.parse_args()
+    return args
+
+
+@app.get("/", status_code=status.HTTP_200_OK)
+@authenticate
+async def root() -> Dict[str, str]:
+    """The api has 2 routes: compile and inference
+
+    Returns:
+        Dict[str, str]: {"message": "The api has 2 routes: compile and inference"}
+    """
+    return {"message": "The api has 2 routes: compile and inference"}
+
+
+@app.post("/compile")
+@authenticate
+async def compile_wf(request: Request) -> Json:
```

Comment:
Return type should be a strongly typed model instead of just Json. Look into using pydantic to enforce types

Reply:
Camilo is looking into Pydantic types. Does that need to be done for this PR, or can we add Pydantic later?

Reply:
If you guys want to merge this quickly it can be done later.